.circleci/config.yml

```diff
@@ -30,12 +38,35 @@ jobs:
       - save_cache:
           paths:
             ./venv/
-          key: v1-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "setup.py" }}-{{ checksum "requirements.txt" }}
+          key: v1-{{ arch }}-{{ .Environment.CIRCLE_PROJECT }}-{{ checksum "setup.py" }}-{{ checksum "requirements.txt" }}
       - run:
           name: Pytest
           command: |
             ./venv/bin/pytest -v tests/
-
+  publish:
+    docker:
+      - image: circleci/python:2.7
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-{{ arch }}-{{ .Environment.CIRCLE_PROJECT }}-{{ checksum "setup.py" }}-{{ checksum "requirements.txt" }}
+      - run:
+          name: verify git tag vs. version
+          command: |
+            python -m venv venv || virtualenv venv
+            . venv/bin/activate
+            python setup.py verify
+      - run:
+          name: create packages
+          command: |
+            python setup.py sdist
+            python setup.py bdist_wheel
+      - run:
+          name: upload to pypi
+          command: |
+            . venv/bin/activate
+            ./venv/bin/twine upload --config .pypirc -r pypi dist/* --password $PYPI_PASSWORD_PROD
```

Comment:
nit: since you're calling `./venv/bin/twine` you don't need to activate the venv before hand (alternatively you could omit `./venv/bin/` and keep the activate call)